### ISSUE TITLE
Add config for compressing options before chrome.sync.set

### ISSUE THREAD

**Issue by mjmaurer** [fregante/webext-options-sync#31]
Thanks for all these awesome webext libraries!

The [sync.set](https://developers.chrome.com/extensions/storage) serialization isn't the most transparent, but from what it sounds like it might be something like JSON.stringify

The storage.sync capacity is 100Kb so I think the smaller the better here.

Could use something like lz-string + JSON.stringify to seriazlize + compress / decompress + deserialize the options map

**Comment by fregante** [comment]
> The [sync.set](https://developers.chrome.com/extensions/storage) serialization isn't the most transparent, but from what it sounds like it might be something like JSON.stringify

Indeed, it looks like that's what they're using:

```js
chrome.storage.sync.getBytesInUse(console.log)
// -> 4256

chrome.storage.sync.get(data => console.log(JSON.stringify(data).length))
// -> 4261
```

This library is pretty small: https://bundlephobia.com/result?p=lz-string@1.4.4

I tried compressing my Refined GitHub options and that number went down to 850 bytes, which is pretty good. It should probably be enabled by default.

The compression takes 7ms on RunKit so if implemented, and the decompression about 2ms.

**Comment by mjmaurer** [comment]
sweet! feel free to assign to me. I think I'll have some time

**Comment by fregante** [comment]
Sure, if you have time it would be great. It should probably be enabled by default but it should still be able to **read** the uncompressed options to allow an easy migration of existing users. Then the options can be compressed on the successive _save_

**Comment by fregante** [comment]
Actually it's best to wait until #28 is merged. Could you try that branch in your extension? Lots of changes coming

**Comment by jroehl** [comment]
Hi, first and foremost - thanks for this awesome little package! 👍
As I need this feature for another project - I implemented the compression in #40.

I am using [compressToEncodedURIComponent](https://pieroxy.net/blog/pages/lz-string/index.html) - as I found that the compression rate was actually higher for js objects. 

What do you think?

**Comment by fregante** [comment]
Fixed by #40 

Available in v1.2.0
